File: leetcode/python/000217.contains-duplicate.py
class Solution:
    # Time complexity: O(n)
    # Space complexity: O(n)
    def containsDuplicate(self, nums: list[int]) -> bool:
        map: dict[int, bool] = {}

        for num in nums:
            if num in map:
                return True

            map[num] = True

        return False

File: leetcode/go/000217.contains_duplicate.go
package main

// Time complexity: O(n)
// Space complexity: O(n)
func containsDuplicate(nums []int) bool {
	set := make(map[int]struct{}, len(nums))

	for _, n := range nums {
		if _, ok := set[n]; ok {
			return true
		}
		set[n] = struct{}{}
	}

	return false
}
